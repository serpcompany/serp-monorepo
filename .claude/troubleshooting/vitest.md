https://vitest.dev/guide/common-errors.html