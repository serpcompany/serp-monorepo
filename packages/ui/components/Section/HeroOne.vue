<script setup lang="ts">
  defineProps<{
    title: string;
    subtitle?: string;
  }>();
</script>

<template>
  <section class="py-20">
    <h1 class="md:text-center">{{ title }}</h1>
    <p
      v-if="subtitle"
      data-testid="subtitle"
      class="sm:text-md text-left text-base md:text-center"
    >
      {{ subtitle }}
    </p>
  </section>
</template>
