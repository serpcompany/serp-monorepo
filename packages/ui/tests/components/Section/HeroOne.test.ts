import { describe, expect, it } from 'vitest'
import HeroOne from '../../../components/Section/HeroOne.vue'
import ComponentRender from '../../componentRender'

describe('sectionHeroOne Snapshot', () => {
  it.each([
    ['with title only', { props: { title: 'Welcome to HeroOne' } }],
    [
      'with title and subtitle',
      { props: { title: 'Welcome to HeroOne', subtitle: 'This is a subtitle' } },
    ],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(`HeroOne ${desc}`, options, HeroOne)
      expect(html).toMatchSnapshot()
    },
  )
})
