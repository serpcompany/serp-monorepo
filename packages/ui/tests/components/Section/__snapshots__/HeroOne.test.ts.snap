// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sectionHeroOne Snapshot > renders with title and subtitle correctly 1`] = `
<section class="py-20">
  <h1 class="md:text-center">
    Welcome to HeroOne
  </h1>
  <p class="
    md:text-center
    sm:text-md
    text-base
    text-left
  ">
    This is a subtitle
  </p>
</section>
`;

exports[`sectionHeroOne Snapshot > renders with title only correctly 1`] = `
<section class="py-20">
  <h1 class="md:text-center">
    Welcome to HeroOne
  </h1>
  <!-- v-if -->
</section>
`;
